### ISSUE TITLE
Error at compare stage of EV couplings complex:  Unable to download and acess files from SIFT database

### ISSUE THREAD

**Issue by sudhabioinfo87** [debbiemarkslab/EVcouplings#290]
The configuration file runs till the coupling stage of the workflow. I guess there is a problem in downloading the SIFTS files. 
Pls find below the error message:
/home/sudha/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py:37: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  LARGE_SPARSE_SUPPORTED = LooseVersion(scipy_version) >= '0.14.0'
/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/visualize/parameters.py:137: RuntimeWarning: divide by zero encountered in log2
  B = -fi * np.log2(fi)
/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/visualize/parameters.py:137: RuntimeWarning: invalid value encountered in multiply
  B = -fi * np.log2(fi)
Traceback (most recent call last):
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 1566, in ftp_open
    fw = self.connect_ftp(user, passwd, host, port, dirs, req.timeout)
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 1588, in connect_ftp
    persistent=False)
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 2408, in __init__
    self.init()
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 2417, in init
    self.ftp.connect(self.host, self.port, self.timeout)
  File "/home/sudha/anaconda3/lib/python3.7/ftplib.py", line 154, in connect
    source_address=self.source_address)
  File "/home/sudha/anaconda3/lib/python3.7/socket.py", line 728, in create_connection
    raise err
  File "/home/sudha/anaconda3/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sudha/anaconda3/bin/evcouplings_runcfg", line 10, in <module>
    sys.exit(app())
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/utils/pipeline.py", line 583, in app
    outcfg = run(**kwargs)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/utils/pipeline.py", line 570, in run
    return execute_wrapped(**config)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/utils/pipeline.py", line 508, in execute_wrapped
    outcfg = execute(**config)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/utils/pipeline.py", line 187, in execute
    outcfg = runner(**incfg)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/compare/protocol.py", line 1250, in run
    return PROTOCOLS[kwargs["protocol"]](**kwargs)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/compare/protocol.py", line 961, in complex
    outcfg, first_sifts_map = _identify_monomer_structures("first", outcfg, first_aux_prefix)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/compare/protocol.py", line 947, in _identify_monomer_structures
    prefix=aux_prefix
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/compare/protocol.py", line 154, in _identify_structures
    kwargs["sifts_sequence_db"]
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/compare/sifts.py", line 268, in __init__
    self._create_mapping_table(sifts_table_file)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/compare/sifts.py", line 334, in _create_mapping_table
    get_urllib(SIFTS_URL, temp_download_file)
  File "/home/sudha/anaconda3/lib/python3.7/site-packages/evcouplings/utils/system.py", line 300, in get_urllib
    with urllib.request.urlopen(url) as r, open(output_path, 'wb') as f:
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 1584, in ftp_open
    raise exc.with_traceback(sys.exc_info()[2])
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 1566, in ftp_open
    fw = self.connect_ftp(user, passwd, host, port, dirs, req.timeout)
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 1588, in connect_ftp
    persistent=False)
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 2408, in __init__
    self.init()
  File "/home/sudha/anaconda3/lib/python3.7/urllib/request.py", line 2417, in init
    self.ftp.connect(self.host, self.port, self.timeout)
  File "/home/sudha/anaconda3/lib/python3.7/ftplib.py", line 154, in connect
    source_address=self.source_address)
  File "/home/sudha/anaconda3/lib/python3.7/socket.py", line 728, in create_connection
    raise err
  File "/home/sudha/anaconda3/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
urllib.error.URLError: <urlopen error ftp error: ConnectionRefusedError(111, 'Connection refused')>
Segmentation fault (core dumped)

Pls find attached the config file. Only the path to SIFTS file has been provided. The file map and db are empty filenames. I was hoping that Evcouplings will automatically download the required SIFTS files and paste in the file path I had provided in the config file as follows:
    sifts_mapping_table: /home/sudha/WORK/programs/evcouplings_support_programs/sifts_map/map
    sifts_sequence_db: /home/sudha/WORK/programs/evcouplings_support_programs/sifts_seqdb/db
 
It will be great, if you could please fix this problem. 
[sample_config_complex.txt](https://github.com/debbiemarkslab/EVcouplings/files/11247957/sample_config_complex.txt)

**Comment by thomashopf** [comment]
Can you please try to run this again with the latest version of EVcouplings directly from the develop branch here on github? There have been some fixes recently related to retrieval of sequences for the SIFTS tables.

Additionally, please check your firewall settings, it might be the case that FTP connections are blocked.

**Comment by thomashopf** [comment]
Assuming as resolved, please reopen if necessary

**Comment by sudhabioinfo87** [comment]
Dear EVcouplings team,

Thanks a lot for the reply. Sorry. I could'nt reply earlier.
I had to work on other sections of my project. I was racing with time.
I was in touch with Daniel Ritter from your group.
He had helped me with troubleshooting.
I tried out his suggestion .It worked !!!
I will try to run  batch jobs on the server.
I will get in touch if I have problems installing on the server.

Thanks again,
Dr. Sudha Govindarajan.
